fhir2dataset/parser.py

```diff
@@ -0,0 +1,193 @@
+            assert alias in self.__from.keys(), (
+                f"The {alias} alias used in the {join_condition} join must have been referenced "
+                "behind a FROM, INNER JOIN, PARENT JOIN or CHILD JOIN. "
+                f"The only aliases referenced are here: {self.__from}"
+            )
+            if join_rule in ["id", "id ", " id", " id "]:
+                if alias_child:
+                    raise ValueError
+                alias_child = alias
+            else:
+                if alias_parent or searchparam_parent:
+                    raise ValueError
+                alias_parent = alias
+                searchparam_parent = join_rule
+
+        return alias_parent, searchparam_parent, alias_child
+
+    def __where_parser(self, string):
+        item_parsed = re.split(self.__split_mask_where, string)
+        # print("\nWHERE")
+        # print(f"item_parsed: {item_parsed}")
+        for where_condition in item_parsed:
+            condition_parsed = re.split(self.__split_mask_where_condition, where_condition)
+            assert len(condition_parsed) == 3
+            alias, where_rule = re.split(r"\.", condition_parsed[0], 1)
+            value = condition_parsed[2]
+            if condition_parsed[1] in PREFIX:
+                prefix = condition_parsed[1].replace(" ", "")
+                self.__where[alias][where_rule] = {prefix: value}
+            else:
+                self.__where[alias][where_rule] = value
+
+    def __init_clauses(self):
+        BASIC_CLAUSES = OrderedDict(
+            {
+                "FROM": self.__from_parser,
+                "INNER JOIN": self.__inner_join_parser,
+                "CHILD JOIN": self.__child_join_parser,
+                "PARENT JOIN": self.__parent_join_parser,
+                "SELECT": self.__select_parser,
+                "WHERE": self.__where_parser,
+            }
+        )
+        for clause, function in BASIC_CLAUSES.items():
+            self.CLAUSES[f" {clause} "] = function
+            self.CLAUSES[f"{clause} "] = function
+            self.CLAUSES[f" {clause}"] = function
+            self.CLAUSES[clause] = function
+
+    def __init_split_mask(self):
+        self.__split_mask_clauses = self.__create_mask_multiple_separators(self.CLAUSES.keys())
+        self.__split_mask_from = " AS "
+        self.__split_mask_select = " , |, "
+        self.__split_mask_from_join = " ON "
+        self.__split_mask_join = " = "
+        self.__split_mask_where = " AND "
+        self.__split_mask_where_condition = self.__create_mask_multiple_separators(PREFIX + [" = "])
+
+    def __create_mask_multiple_separators(self, separators, keep_separators: bool = True):
+        split_mask_multiple = ""
+        for separator in separators:
+            if split_mask_multiple:
+                split_mask_multiple += f"|{separator}"
+            else:
+                split_mask_multiple += f"{separator}"
```

Comment:
```suggestion
        split_mask_multiple = '|'.join(separators)
```

Reply:
I even wonder if a dedicated function is needed? If we can make this function a one liner we won't need it!

Reply:
Great idea! I've removed the useless function